src/Approximations/template_directions.jl

```diff
@@ -350,11 +350,38 @@ function Base.iterate(od::OctDirections{N, SparseVector{N, Int}}, state::Tuple)
     return (copy(vec), (vec, i, j))
 end
 
+function Base.iterate(od::OctDirections{N, SparseVector{N, Int}}, state::Tuple) where {N}
+    _iterate_state(od, state)
+end
+
 function Base.iterate(od::OctDirections{N, SparseVector{N, Int}}, state::Int) where {N}
     # continue with box directions
     return iterate(BoxDirections{N, SparseVector{N, Int}}(od.n), state)
 end
 
+# ----------------------------------
+# implementation with regular arrays
+# ----------------------------------
+function Base.iterate(od::OctDirections{N, Vector{N}}) where {N}
```

Comment:
I guess these methods can also serve as fallback implementation for any vector type (by removing `Vector{N}`).

Reply:
one case that i would be interested is immutable vector types such as static arrays, but they would need a new implementation.